### PR TITLE
Update trigger docs

### PR DESCRIPTION
**Pull request by harrietgrace**
Adding some words about access control to the Trigger step doc re: https://github.com/buildkite/docs/issues/102

I'm not sure if these additions are super accurate - @ticky would you mind having a quick squiz and let me know if anything needs to be added or changed? 😊

### CODE REVIEW

File: pages/pipelines/trigger_step.md.erb [hunk 1]
```diff
@@ -9,6 +9,12 @@ steps:
   - trigger: "another-pipeline"
 ```
 
+The author of the initial build will be the author of all builds created from a trigger step. 
+
+If you have [Teams](/docs/pipelines/permissions) enabled in your organisation, ensure that this author has Build permission on all of the pipelines that will be triggered by this step.
+
+If the author does not have the required access, the initial build will fail, and builds on subsequent pipelines will not be triggered.
```
Comment (ticky) [inline pages/pipelines/trigger_step.md.erb]: This is good, but it’s missing the requirement that the author be part of the org!

Maybe something like

>Any builds created by a trigger step will have the same author as the initiating build.
>
>The author of the build containing the trigger step must be a member of your organisation with a verified email address in order to trigger further builds. If you have [Teams](/docs/pipelines/permissions) enabled in your organisation, the author must have Build permission on the pipelines that will be triggered by this step.
>
>If the author does not have the required access, the trigger step will fail, and the requested builds are not created.

**Comment by harrietgrace** [comment]
Thanks for the help @ticky! I added in the bit about orgs, but it was feeling a bit wordy. This bit is kind of important, so we don't want people skipping over the paragraph. To make it a bit punchier, I changed it to have dot points. What do you think? It doesn't read as smoothly, but I think it highlights the vital information better.

File: pages/pipelines/trigger_step.md.erb [hunk 2]
```diff
@@ -9,6 +9,18 @@ steps:
   - trigger: "another-pipeline"
 ```
 
+All builds created by a trigger step will be authored by the same user that created the initial build. This user must:
```
Comment (ticky) [inline pages/pipelines/trigger_step.md.erb]: “will be authored by” still feels very awkward to me

File: pages/pipelines/trigger_step.md.erb [hunk 3]
```diff
@@ -9,6 +9,18 @@ steps:
...
+
+* be a member of your organisation
+* have a verified email address
+
+If you have [Teams](/docs/pipelines/permissions) enabled in your organisation, the authoring user must also have:
+
+* 'Build' permission on *every* pipeline that will be triggered
+
+If the author does not have the required access, the initial build will fail, and builds on subsequent pipelines will not be triggered.
```
Comment (ticky) [inline pages/pipelines/trigger_step.md.erb]: “initial” build feels like kind of confusing wording

File: pages/pipelines/trigger_step.md.erb (hunk 2 above)
Comment (lox) [reply pages/pipelines/trigger_step.md.erb]: "All builds created by a trigger step will be created by the user that triggered the original build, or the user that unblocked the last preceding block step. That user must be:"

Comment (lox) [reply pages/pipelines/trigger_step.md.erb]: Possibly with "If the build was triggered by a commit, then the user will be inferred from the commit author"?

File: pages/pipelines/trigger_step.md.erb (hunk 3 above)
Comment (lox) [reply pages/pipelines/trigger_step.md.erb]: "If the author doesn't have sufficient access to the pipeline being triggered, the trigger step will fail and cause the original build to fail as well"?

**Comment by harrietgrace** [comment]
Eep, this pr really slipped through the cracks for me. Thanks for the feedbacks @ticky and @lox. 
I removed the 'will be authored by' and 'inital' bits, how does it feel with the revisions?

**Comment by lox** [comment]
Sounds good to me!

**Comment by harrietgrace** [comment]
Going to merge this one - happy to do a V2 sometime in the future if people are getting confused with the wording.
